src/mesido/workflows/io/write_output.py

```diff
@@ -954,19 +954,23 @@ def _name_to_asset(name):
                 *self.energy_system_components.get("heat_source", []),
                 *self.energy_system_components.get("ates", []),
                 *self.energy_system_components.get("heat_buffer", []),
+                *self.energy_system_components.get("heat_pump", []),
             ]:
                 asset = _name_to_asset(name)
                 asset_placement_var = self._asset_aggregation_count_var_map[name]
                 placed = np.round(results[asset_placement_var][0]) >= 1.0
                 max_size = results[self._asset_max_size_map[name]][0]
 
-                if asset in self.energy_system_components.get("heat_buffer", []):
+                if asset.name in self.energy_system_components.get("heat_buffer", []):
                     asset.capacity = max_size
                     asset.volume = max_size / (
                         parameters[f"{name}.cp"]
                         * parameters[f"{name}.rho"]
                         * parameters[f"{name}.dT"]
                     )
+                elif asset.name in self.energy_system_components.get("heat_pump", []):
+                    # Note: Electrical capacity and not the heat capacity
+                    asset.power = max(results["HeatPump_cc7b.Power_elec"])
```

Comment:
The ates doesn't have a power, so shouldn't a different attribute be updated?

Reply:
ATES charge rates are now updated